tensorflow_asr/models/transducer/transducer.py

```diff
@@ -604,14 +604,14 @@ def recognize_beam(self, inputs: Dict[str, tf.Tensor], lm: bool = False):
         """
         RNN Transducer Beam Search
         Args:
-            features (tf.Tensor): a batch of padded extracted features
+            inputs (Dict[str, tf.Tensor]): Input dictionary containing "inputs" and "inputs_length"
             lm (bool, optional): whether to use language model. Defaults to False.
 
         Returns:
             tf.Tensor: a batch of decoded transcripts
         """
         encoded = self.encoder(inputs["inputs"], training=False)
-        encoded_length = math_util.get_reduced_length(inputs["inputs_length"], self.time_reduction_factor)
+        encoded_length = inputs["inputs_length"]
```

Comment:
@ebraraktas Hmm, why did you remove the `get_reduced_length`?

Reply:
That [commit message](https://github.com/TensorSpeech/TensorFlowASR/pull/204/commits/95cdd0e350febad59668420ab286e35dd0b7bd02) has similar but a little bit detailed explanation

Reply:
if you remove `get_reduced_length` from the `recognize_beam`, the `test.py` with beam search will wrong because the `ASRDataset` creates inputs with the original, "not reduced" length 😄

Reply:
But the thing is if you reduce `inputs["inputs_length"]` again it will be reduced 2 times. I have tested recogniton of `examples/demonstration/wavs/2033-164915-0001.flac`. `inputs["inputs_length"]` is 224 for that audio. If you call `get_reduced_length` again, it will return 56, and recognition is done for one fourth of the audio. Maybe using `encoded_length = tf.expand_dims(encoded.shape[1], 0)` fixes.

Reply:
Sorry, I see what you meant now. This suggestion seems fine. I will fix it.

> Ah no, the `inputs_length` is the length of the audio feature, but the `encoded_length` is the length of the result of the encoder. I suggest we remove `get_reduced_length` from [this line](https://github.com/TensorSpeech/TensorFlowASR/pull/204/files#diff-51ca0fe66fb1885fac577d2861e52e009ca8dac4616b93aeaad14953d2eed01bR74) and keep the `get_reduced_length` in `recognize_beam`